crawlerr/services/rag_service.py

```
@@ -73,6 +162,215 @@ def chunk_text(self, text: str, metadata: Dict[str, Any] = None) -> List[Dict[st
+        search_text = full_text[start_char:min(end_char + 200, len(full_text))]
+        ideal_end = end_char - start_char
+        
+        # Try boundary detectors in order of preference
+        boundary_finders = [
+            (self._find_paragraph_boundary, "paragraph"),
+            (self._find_sentence_boundary, "sentence"),
+            (self._find_line_boundary, "line"),
+            (self._find_word_boundary, "word")
+        ]
+        
+        for finder, boundary_type in boundary_finders:
```

Ruff (0.12.2)

Loop control variable `boundary_type` not used within loop body

Rename unused `boundary_type` to `_boundary_type`

(B007)